cdap-cli/src/main/java/co/cask/cdap/cli/CLIMain.java

```diff
@@ -248,10 +265,63 @@ public static void main(String[] args) {
         cliMain.tryAutoconnect();
         cliMain.updateCLIPrompt(cliConfig.getClientConfig());
 
-        if (commandArgs.length == 0) {
+        boolean ranScript = false;
+        boolean brokeFromScript = false;
+        if (launchOptions.getScriptFile() != null) {
+
+          ranScript = true;
+
+          ConnectionConfig connectionConfig = clientConfig.getConnectionConfig();
+          URI baseURI = connectionConfig.getURI();
+          URI uri = baseURI.resolve("/" + connectionConfig.getNamespace());
+          String cliPrompt = "\ncdap (" + uri + ")> ";
+
+          BufferedReader br = null;
+          try {
+            String line;
+            br = new BufferedReader(new FileReader(launchOptions.getScriptFile()));
+            // br = Files.newReader(launchOptions.getScriptFile(), Charset.UTF_8);
+            while ((line = br.readLine()) != null) {
+              String ln = line.trim();
+              String[] lnArr = ln.split("#");
```

Comment:
This comment feature should also be supported in other cases: Entering a command in interactive mode, and entering command from command line (e.g. `cdap-cli.sh deploy app purchase.jar`)